bin/liquidator/src/rebalance.rs

```diff
@@ -147,29 +159,34 @@ impl Rebalancer {
         let mut jobs = vec![full_route_job, direct_quote_route_job];
 
         for in_token_index in &self.config.alternate_jupiter_route_tokens {
-            let in_token = self.mango_client.context.token(*in_token_index);
-            // For the alternate output routes we need to adjust the in amount by the token price
-            let in_price = self
-                .account_fetcher
-                .fetch_bank_price(&in_token.first_bank())?;
-            let in_amount = (I80F48::from(in_amount_quote) / in_price)
-                .ceil()
-                .to_num::<u64>();
+            let (alt_mint, alt_in_amount) =
+                self.get_alternative_token_amount(in_token_index, in_amount_quote)?;
             let direct_route_job =
-                self.jupiter_quote(in_token.mint, output_mint, in_amount, true, jupiter_version);
+                self.swap_quote(alt_mint, output_mint, alt_in_amount, true, jupiter_version);
             jobs.push(direct_route_job);
         }
 
-        let mut results = futures::future::join_all(jobs).await;
-        let full_route = results.remove(0)?;
-        let alternatives = results.into_iter().filter_map(|v| v.ok()).collect_vec();
+        if self.config.use_sanctum && self.is_lst(output_mint) {
+            // go through an alternative token
+            for in_token_index in &self.config.alternate_sanctum_route_tokens {
+                let (alt_mint, alt_in_amount) =
+                    self.get_alternative_token_amount(in_token_index, in_amount_quote)?;
+                let sanctum_alt_route_job = self.swap_quote(
+                    alt_mint,
+                    output_mint,
+                    alt_in_amount,
+                    true,
+                    swap::Version::Sanctum,
+                );
+                jobs.push(sanctum_alt_route_job);
+            }
+        }
+
+        let results = futures::future::join_all(jobs).await;
```

Comment:
future pr: instead of configuring individual timeouts, it could be good to give a general quote timeout that applies here